src/py/flwr/cli/install.py

```diff
@@ -0,0 +1,208 @@
+    ``flwr-dir`` is equal to:
+
+        - ``$FLWR_HOME/`` if ``$FLWR_HOME`` is defined
+        - ``$XDG_DATA_HOME/.flwr/`` if ``$XDG_DATA_HOME`` is defined
+        - ``$HOME/.flwr/`` in all other cases
+    """
+    if source is None:
+        source = Path(typer.prompt("Enter the source FAB file"))
+
+    source = source.resolve()
+    if not source.exists() or not source.is_file():
+        typer.secho(
+            f"❌ The source {source} does not exist or is not a file.",
+            fg=typer.colors.RED,
+            bold=True,
+        )
+        raise typer.Exit(code=1)
+
+    if source.suffix != ".fab":
+        typer.secho(
+            f"❌ The source {source} is not a `.fab` file.",
+            fg=typer.colors.RED,
+            bold=True,
+        )
+        raise typer.Exit(code=1)
+
+    install_from_fab(source, flwr_dir)
+
+
+def install_from_fab(
+    fab_file: Path, flwr_dir: Optional[Path], skip_prompt: bool = False
+) -> None:
+    """Install from a FAB file after extracting and validating."""
+    with tempfile.TemporaryDirectory() as tmpdir:
+        with zipfile.ZipFile(fab_file, "r") as zipf:
+            zipf.extractall(tmpdir)
+            tmpdir_path = Path(tmpdir)
+            info_dir = tmpdir_path / ".info"
+            if not info_dir.exists():
+                typer.secho(
+                    "❌ FAB file has incorrect format.",
+                    fg=typer.colors.RED,
+                    bold=True,
+                )
+                raise typer.Exit(code=1)
+
+            content_file = info_dir / "CONTENT"
+
+            if not content_file.exists() or not _verify_hashes(
+                content_file.read_text(), tmpdir_path
+            ):
+                typer.secho(
+                    "❌ File hashes couldn't be verified.",
+                    fg=typer.colors.RED,
+                    bold=True,
+                )
+                raise typer.Exit(code=1)
+
+            shutil.rmtree(info_dir)
+
+            validate_and_install(tmpdir_path, fab_file.stem, flwr_dir, skip_prompt)
+
+
+def validate_and_install(
+    project_dir: Path,
+    fab_name: str,
+    flwr_dir: Optional[Path],
+    skip_prompt: bool = False,
+) -> None:
+    """Validate TOML files and install the project to the desired directory."""
+    config, _, _ = load_and_validate(project_dir / "pyproject.toml", check_module=False)
+
+    if config is None:
+        typer.secho(
+            "❌ Invalid config inside FAB file.",
+            fg=typer.colors.RED,
+            bold=True,
+        )
+        raise typer.Exit(code=1)
+
+    username = config["flower"]["publisher"]
+    project_name = config["project"]["name"]
+    version = config["project"]["version"]
+
+    if fab_name != f"{username}.{project_name}.{version.replace('.', '-')}":
+        typer.secho(
+            "❌ FAB file has incorrect name, it should be "
+            "`<username>.<project_name>.<version>.fab`.",
```

Comment:
```suggestion
            "❌ FAB file has incorrect name. The file name must follow the format "
            "`<username>.<project_name>.<version>.fab`.",
```

Reply:
@danieljanes I was planning to remove this check in the future as it is not relevant for when we are working with raw bytes and also I think it should be possible to install the FAB even though the filename is something else. Wdyt? I can also add a new parameter like `check_filename` that would be set to `True` when using the CLI